raster/src/main/scala/geotrellis/raster/interpolation/README.md

```diff
@@ -0,0 +1,140 @@
+    
+    val krigingVal: Tile =
+        points.ordinaryKriging(rasterExtent, 5000, sv)
+        
+    /**
+      * The user can also do Ordinary Kriging using :
+      * points.ordinaryKriging(rasterExtent)
+      * points.ordinaryKriging(rasterExtent, bandwidth)
+      * points.ordinaryKriging(rasterExtent, Semivariogram)
+      * points.ordinaryKriging(rasterExtent, bandwidth, Semivariogram)
+      */
+
+It is belongs to the class of Simple Spatial Prediction Models.
+
+This method differs from the Simple Kriging appraoch in that, the constant mean is assumed to be unknown and is estimated within the model.
+
+    mu(s) = mu              unknown; s belongs to R
+    cov[eps(s), eps(s')]    known; s, s' belongs to R
+
+#### Universal Kriging
+    //Universal kriging, a tile  set with the Kriging prediction per cell is returned
+    
+    val attrFunc: (Double, Double) => Array[Double] = {
+      (x, y) => Array(x, y, x * x, x * y, y * y)
+    }
+      
+    val krigingVal: Tile = 
+        points.universalKriging(rasterExtent, attrFunc, 50, Spherical)
+      
+    /**
+      * The user can also do Universal Kriging using :
+      * points.universalKriging(rasterExtent)
+      * points.universalKriging(rasterExtent, bandwidth)
+      * points.universalKriging(rasterExtent, model)
+      * points.universalKriging(rasterExtent, bandwidth, model)
+      * points.universalKriging(rasterExtent, attrFunc)
+      * points.universalKriging(rasterExtent, attrFunc, bandwidth)
+      * points.universalKriging(rasterExtent, attrFunc, model)
+      * points.universalKriging(rasterExtent, attrFunc, bandwidth, model)
+      */
+
+It is belongs to the class of General Spatial Prediction Models.
+
+This model allows for explicit variation in the trend function (mean function) constructed as a linear function of spatial attributes; with the covariance values assumed to be known. This model computes the prediction using
+
+For example if :
+
+    x(s) = [1, s1, s2, s1 * s1, s2 * s2, s1 * s2]'
+    mu(s) = beta0 + beta1*s1 + beta2*s2 + beta3*s1*s1 + beta4*s2*s2 + beta5*s1*s2
+Here, the "linear" refers to the linearity in parameters (beta).
+
+    mu(s) = x(s)' * beta,   beta unknown; s belongs to R
+    cov[eps(s), eps(s')]    known; s, s' belongs to R
+    
+The `attrFunc` function is the attribute function, which is used for evaluating non-constant spatial trend structures. Unlike the Simple and Ordinary Kriging models which rely only on the residual values for evaluating the spatial structures, the General Spatial Models may be modelled by the user based on the data (viz. evaluating the beta variable to be used for interpolation).
+
+In case the user does not specify an attribute function, by default the function used is a quadratic trend function for Point(s1, s2) :
+
+```mu(s) = beta0 + beta1*s1 + beta2*s2 + beta3*s1*s1 + beta4*s2*s2 + beta5*s1*s2```
+
+General example of a trend function is : 
+
+```mu(s) = beta0 + Sigma[ beta_j * (s1^n_j) * (s2^m_j) ]```
```

Comment:
Thanks for the explanation! Could you perhaps give a use-case example of when someone would not want to use the default attribute function, and they would perhaps use a different attribute function? Giving a real-world example of it might help clarify when it *would* be used, even if it's the case that if you don't know how you'd use it then you probably shouldn't :)